applications/adv_networking_bench/cpp/CMakeLists.txt

```diff
@@ -104,43 +104,55 @@ install(
+    if(NOT result_code EQUAL 0 OR "${${OUTPUT_VAR}}" STREQUAL "")
+      message(STATUS "[ANO Test] ${MSG}")
+      message(WARNING "'${CMD}' failed (${result_code}) or output was empty.\nError: ${error_output}")
+      message(STATUS "[ANO Test] Skipping advanced networking loopback test")
+      return()
+    else()
+      message(STATUS "[ANO Test] ${MSG} ${${OUTPUT_VAR}}")
+    endif()
+  endmacro()
+
+  # Check NIC addresses
+  message(STATUS "[ANO Test] Checking system compatibility for advanced networking loopback test")
+  execute_and_check("NVIDIA NIC BUS Addresses:" "lspci -nn | grep -iE 'Ethernet controller.*15b3:' | awk '{print $1}'" cx_bus_address_list)
+  string(REGEX REPLACE "\n" ";" cx_bus_address_list "${cx_bus_address_list}")
+  list(LENGTH cx_bus_address_list cx_bus_address_list_count)
+  if(NOT cx_bus_address_list_count EQUAL 2)
+    message(WARNING "[ANO Test] Expected 2 CX NICs, found ${cx_bus_address_list_count}")
+    message(WARNING "[ANO Test] Skipping advanced networking test")
+    return()
+  endif()
+  list(GET cx_bus_address_list 0 tx_bus)
+  list(GET cx_bus_address_list 1 rx_bus)
+  execute_and_check("ETH Interface for ${tx_bus}:" "find /sys/class/net -exec bash -c 'echo {} $(readlink -f {}/device)' \; | grep ${tx_bus} | awk '{print $1}'" tx_if)
+  execute_and_check("MAC Address for ${tx_if}:" "cat ${tx_if}/address" rx_eth_mac)
+
+  # Inject in the test config and add app timeout
   file(READ "${CMAKE_CURRENT_SOURCE_DIR}/../adv_networking_bench_default_tx_rx.yaml" TEST_CONFIG)
-  string(REGEX REPLACE "eth_dst_addr:[^\n]*" "eth_dst_addr: ${rx_eth_mac}" TEST_CONFIG ${TEST_CONFIG})
   string(REGEX REPLACE "# max_duration_ms:[^\n]*" "max_duration_ms: 20000" TEST_CONFIG ${TEST_CONFIG})
-  file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/adv_networking_bench_default_tx_rx_test.yaml" ${TEST_CONFIG})
+  string(REGEX REPLACE "address: [^\n]*interface[^\n]*Rx[^\n]*" "address: ${rx_bus}" TEST_CONFIG ${TEST_CONFIG})
+  string(REGEX REPLACE "address: [^\n]*interface[^\n]*Tx[^\n]*" "address: ${tx_bus}" TEST_CONFIG ${TEST_CONFIG})
+  string(REGEX REPLACE "address: [^\n]*Source[^\n]*Tx[^\n]*" "address: ${tx_bus}" TEST_CONFIG ${TEST_CONFIG})
+  string(REGEX REPLACE "eth_dst_addr:[^\n]*" "eth_dst_addr: ${rx_eth_mac}" TEST_CONFIG ${TEST_CONFIG})
+  file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/adv_networking_bench_dpdk_loopback_test.yaml" ${TEST_CONFIG})
 
-  # Create test
+  # Define the test
   file(COPY "testing/adv_networking_test_harness.py" DESTINATION "testing")
   add_test(NAME adv_networking_bench_dpdk_test
-           COMMAND python3 testing/adv_networking_test_harness.py "./adv_networking_bench adv_networking_bench_default_tx_rx_test.yaml" --avg-throughput-threshold 90 --missed-packets-threshold 0.1
+           COMMAND python3 testing/adv_networking_test_harness.py "./adv_networking_bench adv_networking_bench_dpdk_loopback_test.yaml" --avg-throughput-threshold 90 --missed-packets-threshold 0.1
```

Comment:
I would change the file name to "multi_if_loopback" to indicate we're using multiple interfaces with a cable.

Reply:
ty, done